runtime/onert/api/nnfw/include/nnfw_experimental.h

```diff
@@ -233,9 +228,9 @@ typedef struct nnfw_train_info
   /** optimizer type */
   NNFW_TRAIN_OPTIMIZER opt = NNFW_TRAIN_OPTIMIZER_SGD;
 
-  /** indexes of trainable operations */
-  uint32_t trainble_ops_size = 0;
-  uint32_t trainble_ops_idx[NNFW_TRAINABLE_OPS_MAX_SIZE];
+  /** Number of layers to be trained from the back.
+   *  "0" means that all layers should be trained. Negative value means error. */
+  int32_t num_of_trainable_ops = 0;
```

Comment:
@Samsung/one_onert How about using zero value as a non trainable circle file?  And `-1` means that all layers should be trained.

| | before | suggestion |
|:--:|:--:|:--:|
| -N | Error (non trainable model) | Error |
| -1 | Error (non trainable model) | train whole model |
| 0 | train whole model | non trainable model |
| N | train only as many layers as described | same |

Reply:
Giving place to `-1` looks good to me. 

I did a similar approach in this PR (https://github.com/Samsung/ONE/pull/13261#discussion_r1647574923), <sub>But in my PR,  the meaning of 0 and -1 is reversed</sub> .

Reply:
@jyoungyun Your suggestion looks good to me.

Reply:
@jyoungyun I like this idea. I've introduced also enum with special values of `num_of_trainable_ops` to avoid magic values in many places